driver/docker-container/driver.go

```diff
@@ -40,6 +41,13 @@ type Driver struct {
 	factory      driver.Factory
 	netMode      string
 	image        string
+	memory       string
+	memorySwap   string
+	cpuQuota     int64
+	cpuPeriod    int64
+	cpuShares    int64
+	cpusetCpus   string
+	cpusetMems   string
```

Comment:
I think it might be better if we utilize the equivalent types from the `docker create` command: https://github.com/docker/cli/blob/0e70f1b7b831565336006298b9443b015c3c87a5/cli/command/container/opts.go#L34-L132

This plugin also has `docker/cli` as a dependency so we should be able to utilize the `github.com/docker/cli/opts` package which I think we should use instead so the options here are consistent with the equivalent options.

An alternative is to utilize the method that compose does to parse the options, but it's a bit more [adhoc](https://github.com/compose-spec/compose-go/blob/ba097a5dee2003644e4b172f63afbdb5a613b29c/loader/loader.go#L1326-L1337). The underlying implementation seems to be the same, but we may not want to reuse that section of code mostly because it is not easily reusable outside of a YAML context.

Reply:
Agree, let's avoid adding unnecessary dep. `opts.MemBytes` can be used similar to https://github.com/docker/buildx/blob/e018f8b6fbf746e4ac4f019bc3918c92c67b927a/commands/build.go#L74

Reply:
@jsternberg @crazy-max done, the types have been aligned to `github.com/docker/cli/opts`, and the unnecessary deps have been removed